quickwit-doc-mapping/src/mapper.rs

```diff
@@ -43,95 +44,88 @@ use tantivy::{
 /// - a way to build a tantivy::Query from a SearchRequest
 /// - a way to build a tantivy:Schema
 ///
-pub trait DocMapper: Send + Sync + 'static {
+#[typetag::serde(tag = "type", content = "attributes")]
+pub trait DocMapper: Send + Sync + Debug + DynClone + 'static {
     /// Returns the document built from a json string.
     fn doc_from_json(&self, doc_json: &str) -> Result<Document, DocParsingError>;
     /// Returns the schema.
     fn schema(&self) -> Schema;
     /// Returns the query.
     fn query(&self, _request: SearchRequest) -> Box<dyn Query>;
 }
-// TODO: this is a placeholder, to be removed when it will be implementend in the search-api crate
-pub struct SearchRequest {}
 
-/// A `DocMapperType` describe a set of rules to build a document, query and schema.
-#[derive(Clone, Debug, Serialize, Deserialize)]
-#[serde(tag = "type", content = "config", rename_all = "snake_case")]
-pub enum DocMapperType {
-    /// Default doc mapper which is build from a config file
-    Default(DocMapperConfig),
-    /// All flatten doc mapper which indexes everything in one single field
-    AllFlatten,
-    /// Wikipedia doc mapper
-    Wikipedia,
-}
+clone_trait_object!(DocMapper);
 
-impl TryFrom<&str> for DocMapperType {
-    type Error = String;
-
-    fn try_from(doc_mapper_type_str: &str) -> Result<Self, Self::Error> {
-        match doc_mapper_type_str.trim().to_lowercase().as_str() {
-            "all_flatten" => Ok(Self::AllFlatten),
-            "wikipedia" => Ok(Self::Wikipedia),
-            "default" => Ok(Self::Default(DocMapperConfig::default())),
-            _ => Err(format!(
-                "Could not parse `{}` as valid doc mapper type.",
-                doc_mapper_type_str
-            )),
-        }
-    }
-}
+// TODO: this is a placeholder, to be removed when it will be implementend in the search-api crate
+pub struct SearchRequest {}
 
 /// Build a doc mapper given the doc mapper type.
-pub fn build_doc_mapper(mapper_type: DocMapperType) -> anyhow::Result<Box<dyn DocMapper>> {
-    match mapper_type {
-        DocMapperType::Default(config) => {
+pub fn build_doc_mapper(
```

Comment:
I don't think we need this function at all, do we?

Reply:
I think @fmassot  is in a better position to answer this. I believe there is going to be validation on the config.

Reply:
let's remove it as currently we only need something to build the doc mapper from the type in the cli. So we can put that only in the cli.

Reply:
Why do we need that in the CLI?